Update node syncing content [Fixes #12003] (#13201)
ethereum/ethereum-org-website: public/content/developers/docs/nodes-and-clients/index.md

### Execution layer sync modes {#execution-layer-sync-modes}

> #### Full node syncs

Reviewer: ```suggestion
#### Full node syncs {#full-node-syncs}
```

You may run a full node with a:

- [full sync](#full-sync)
- [fast sync](#fast-sync)

> - [snap sync](#snap-sync)

Reviewer: I'm not sure having a bulleted list with hash links is necessary here... the content isn't very long, might suggest we just jump right into the sync options.

Also, ideally we keep the content tree structure to no deeper than h4 (`####`)... any way we could adjust this so we get straight into the options under "Execution layer sync modes"? like `#### Full node: full sync`, `#### Full node: fast sync`, `#### Full node: snap sync`?

Author: fixed with sentence structure instead

Author: fixed

##### Full sync {#full-sync}

A full sync downloads all blocks (including headers and block bodies) and regenerates the state of the blockchain incrementally by executing every block from genesis. 

- Minimizes trust and offers the highest security by verifying every transaction.
- With an increasing number of transactions, it can take days to weeks to process all transactions.

[Archive nodes](#archive-node) perform a full sync to build (and retain) a complete history of the state changes made by every transaction in every block.

##### Fast sync {#fast-sync}

Like a full sync, a fast sync downloads all blocks (including headers, transactions, and receipts). However, instead of re-processing the historical transactions, a fast sync relies on the receipts until it reaches a recent head, when it switches to importing and processing blocks to provide a full node. 

- Fast sync strategy.

> - Reduces processing demand in favour of bandwith usage.

Reviewer: ```suggestion
- Reduces processing demand in favor of bandwidth usage.
```

**NB** Light sync does not yet work with proof-of-stake Ethereum - new versions of light sync should ship soon!

> [More on light clients](/developers/docs/nodes-and-clients/light-clients/).

Reviewer: This is a convention used commonly on the site, without a trailing hard stop; please revert these

```suggestion
[More on light clients](/developers/docs/nodes-and-clients/light-clients/)
```

### Consensus layer sync modes {#consensus-layer-sync-modes}

#### Optimistic sync {#optimistic-sync}

Optimistic sync is a post-merge synchronization strategy designed to be opt-in and backwards compatible, allowing execution nodes to sync via established methods. The execution engine can _optimistically_ import beacon blocks without fully verifying them, find the latest head, and then start syncing the chain with the above methods. Then, after the execution client has caught up, it will inform the consensus client of the validity of the transactions in the Beacon Chain.

> [More on optimistic syncs](https://github.com/ethereum/consensus-specs/blob/dev/sync/optimistic.md).

Reviewer: It's singular in the other uses, I would leave this
```suggestion
[More on optimistic sync](https://github.com/ethereum/consensus-specs/blob/dev/sync/optimistic.md)
```

#### Checkpoint sync {#checkpoint-sync}

A checkpoint sync, also known as weak subjectivity sync, creates a superior user experience for syncing a Beacon Node. It's based on assumptions of [weak subjectivity](/developers/docs/consensus-mechanisms/pos/weak-subjectivity/) which enables syncing the Beacon Chain from a recent weak subjectivity checkpoint instead of genesis. Checkpoint syncs make the initial sync time significantly faster with similar trust assumptions as syncing from [genesis](/glossary/#genesis-block).

In practice, this means your node connects to a remote service to download recent finalized states and continues verifying data from that point. The third party providing the data is trusted and should be picked carefully.

> More on [checkpoint syncs](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice).

Reviewer: I'm not sure I'm a fan of changing to "syncs" here, but not a hill I'll die on

Author: don't have a rationale other than "it sounds better out loud"!

#### Checkpoint sync {#checkpoint-sync}

A checkpoint sync, also known as weak subjectivity sync, creates a superior user experience for syncing a Beacon Node. It's based on assumptions of [weak subjectivity](/developers/docs/consensus-mechanisms/pos/weak-subjectivity/) which enables syncing the Beacon Chain from a recent weak subjectivity checkpoint instead of genesis. Checkpoint syncs make the initial sync time significantly faster with similar trust assumptions as syncing from [genesis](/glossary/#genesis-block).

In practice, this means your node connects to a remote service to download recent finalized states and continues verifying data from that point. The third party providing the data is trusted and should be picked carefully.

> More on [checkpoint syncs](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice).

Reviewer: @m4sterbunny Hah fair enough... I tend to lean towards consistency with these things... we're using "sync" in all of the headers and "More on snap sync", "More on optimistic sync" for the trailing read more links above... I'd suggest we just go with "More on checkpoint sync"

```suggestion
More on [checkpoint sync](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice)
```

Or if you disagree, please at least remove the trailing hard stop:

```suggestion
More on [checkpoint syncs](https://notes.ethereum.org/@djrtwo/ws-sync-in-practice)
```
